lib/seccomp/unittest/CSystemCallFilterTest.cc

```diff
@@ -0,0 +1,216 @@
+#endif
+
+
+class CNamedPipeWriter : public ml::core::CThread {
+public:
+    CNamedPipeWriter(const std::string& fileName, size_t size)
+        : m_FileName(fileName), m_Size(size) {}
+protected:
+    virtual void run() {
+        // Wait for the file to exist
+        ml::core::CSleep::sleep(SLEEP_TIME_MS);
+
+        std::ofstream strm(m_FileName.c_str());
+        for (size_t i = 0; i < m_Size && strm.good(); ++i) {
+            strm << TEST_CHAR;
+        }
+    }
+
+    virtual void shutdown() {}
+
+private:
+    std::string m_FileName;
+    size_t m_Size;
+};
+
+class CNamedPipeReader : public ml::core::CThread {
+public:
+    CNamedPipeReader(const std::string& fileName) : m_FileName(fileName) {}
+
+    const std::string& data() const { return m_Data; }
+
+protected:
+    virtual void run() {
+        m_Data.clear();
+
+        std::ifstream strm;
+
+        // Try to open the file repeatedly to allow time for the other
+        // thread to create it
+        size_t attempt(1);
+        do {
+            CPPUNIT_ASSERT(attempt++ <= MAX_ATTEMPTS);
+            ml::core::CSleep::sleep(SLEEP_TIME_MS);
+            strm.open(m_FileName.c_str());
+        } while (!strm.is_open());
+
+        static const std::streamsize BUF_SIZE = 512;
+        char buffer[BUF_SIZE];
+        while (strm.good()) {
+            strm.read(buffer, BUF_SIZE);
+            CPPUNIT_ASSERT(!strm.bad());
+            if (strm.gcount() > 0) {
+                // This code deals with the test character we write to
+                // detect the short-lived connection problem on Windows
+                const char* copyFrom = buffer;
+                size_t copyLen = static_cast<size_t>(strm.gcount());
+                if (m_Data.empty() && *buffer == ml::core::CNamedPipeFactory::TEST_CHAR) {
+                    ++copyFrom;
+                    --copyLen;
+                }
+                if (copyLen > 0) {
+                    m_Data.append(copyFrom, copyLen);
+                }
+            }
+        }
+    }
+
+    virtual void shutdown() {}
+
+private:
+    std::string m_FileName;
+    std::string m_Data;
+};
+}
+
+bool systemCall() {
+    return std::system("hostname") == 0;
+}
+
+CppUnit::Test* CSystemCallFilterTest::suite() {
+    CppUnit::TestSuite* suiteOfTests = new CppUnit::TestSuite("CSystemCallFilterTest");
+
+    suiteOfTests->addTest(new CppUnit::TestCaller<CSystemCallFilterTest>(
+        "CSystemCallFilterTest::testSystemCallFilter",
+        &CSystemCallFilterTest::testSystemCallFilter));
+
+    return suiteOfTests;
+}
+
+void CSystemCallFilterTest::testSystemCallFilter() {
+#ifdef Linux
+    std::string release{ml::core::CUname::release()};
+    ml::core::CRegex semVersion;
+    semVersion.init("(\\d)\\.(\\d{1,2})\\.(\\d{1,2}).*");
+    ml::core::CRegex::TStrVec tokens;
+    CPPUNIT_ASSERT(semVersion.tokenise(release, tokens));
+    // Seccomp is available in kernals since 3.5
+    if (std::stoi(tokens[0]) < 3 || std::stoi(tokens[1]) < 5) {
```

Comment:
This means the test won't run on kernel version 4.4, as used on Ubuntu 16.04.  The second condition should only be checked if the first number equals 3.